pandas/core/ops.py

```diff
@@ -136,6 +137,62 @@ def maybe_upcast_for_op(obj):
     return obj
 
 
+class CompWrapper(object):
```

Comment:
Haven't looked at implementation in detail but can you add a docstring for this class to describe utility and parameters? Also what is the reasoning for going this route instead of say a meta class?